contracts/ics721-escrow/src/contract.rs

```diff
@@ -0,0 +1,140 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    to_binary, Binary, Deps, DepsMut, Empty, Env, MessageInfo, Response, StdResult, WasmMsg,
+};
+use cw2::set_contract_version;
+
+use crate::{
+    error::ContractError,
+    msg::{ExecuteMsg, InstantiateMsg, QueryMsg},
+    state::{ADMIN_ADDRESS, CHANNEL_ID},
+};
+
+const CONTRACT_NAME: &str = "crates.io:ics721-escrow";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    let admin_address = deps.api.addr_validate(&msg.admin_address)?;
+    ADMIN_ADDRESS.save(deps.storage, &admin_address)?;
+    CHANNEL_ID.save(deps.storage, &msg.channel_id)?;
```

Comment:
Quick question, why CHANNEL_ID is needed in the escrow?
Its not being used except for saving it and query it.

Reply:
Say you knew a NFT was in an escrow contract and wanted to figure out what channel you needed to send it back on in order to release it, then you might want to use this to work backwards. :)